hal/src/electron/modem/mdm_hal.cpp

```diff
@@ -960,13 +972,20 @@ bool MDMParser::powerOn(const char* simpin)
             goto failure;
         }
         else if (i==4 && (RESP_OK != ret) && !retried_after_reset) {
-            retried_after_reset = true; // only perform reset & retry sequence once
-            i = 0;
-            if(!powerOff())
-                reset();
-            /* Power on the modem and perform basic initialization again */
-            if (!_powerOn())
+            if (_dev.sim == SIM_MISSING) {
```

Comment:
On Electron it uses physical SIM. If `_powerOn()` succeeded, which means that the modem is responsive well enough, while it just doesn't detect a physical SIM. In this case, even if we perform a `powerOff()` and then `reset()` and then `powerOn()`, the physical SIM is not there still. Unless user can insert a physical SIM when **seeing** the log in real real quick, then the reset may help, otherwise, I think it's just a waste of time that prevents the device from doing other things afterward and the log is messy.

Reply:
Let's revert this. It's a change in the behavior and a missing SIM may also be a temproary condition due to simply loose contacts or something like that. I would even propose we leave `powerOn()` here as-is.

Reply:
Should we keep the changes to `_powerOn()`. As for SARA-U2, the first low level pulse on PWR_ON may have powered on the modem, while the following 150ms low level pulse might power off the modem immediately.

Reply:
It shouldn't though. The power-off condition on SARA U2 is low pulse over 1s.

Reply:
![image](https://user-images.githubusercontent.com/388478/97603874-726dea80-1a3f-11eb-9134-83ed43fa2122.png)

Reply:
I see. Then why just give it the second low level pulse for all Electron modem types? As for SARA-U2, it has to walk through the second low level pulse any way.

Reply:
We can agree that ideally we should wait properly, but it is what it is. It works and doesn't have any negative consequences as of right now.

Reply:
Sounds reasonable. I'll revert the changes to the power on sequence.